packages/taquito/src/operations/operation-emitter.ts

```diff
@@ -81,17 +83,33 @@ export abstract class OperationEmitter {
     const publicKeyHash = pkh ? pkh : await this.signer.publicKeyHash();
     let counterPromise: Promise<string | undefined> = Promise.resolve(undefined);
 
+    // initializes a currentVotingPeriod if the operation is a ballot op
+    let currentVotingPeriodPromise: Promise<VotingPeriodBlockResult | undefined> =
+      Promise.resolve(undefined);
+    ops.find(async (op) => {
+      if (op.kind === 'ballot') {
+        try {
+          currentVotingPeriodPromise = this.rpc.getCurrentPeriod();
+        } catch (e) {
+          throw new RPCResponseError(
+            `Failed to get the current voting period index: ${JSON.stringify(e)}`
+          );
+        }
+      }
+    });
+
     for (let i = 0; i < ops.length; i++) {
       if (isOpRequireReveal(ops[i]) || ops[i].kind === 'reveal') {
         counterPromise = this.context.readProvider.getCounter(publicKeyHash, 'head');
         break;
       }
     }
 
-    const [hash, protocol, headCounter] = await Promise.all([
+    const [hash, protocol, headCounter, currentVotingPeriod] = await Promise.all([
       blockHashPromise,
       blockProtoPromise,
       counterPromise,
+      currentVotingPeriodPromise,
```

Comment:
my own learning

is this a more efficient way of awaiting multiple values? that you sort them all into a promise all and grab the values needed?

Reply:
kind of, it's just an older way to handle asynchronous data before async/await. In this case I stuck to the pattern that's already available.